monetio/models/_camx_mm.py

```diff
@@ -416,23 +439,28 @@ def _calc_midlayer_height_agl(dset):
         "COL",
     ), "Check dims of z, should be [TSTEP, LAY, ROW, COL]"
 
-    mid_layer_height = np.array(dset["z"])  # height in the layer upper interface of each layer
+    if "z" in list(dset.variables):
+        height = "z"
+    elif "ZGRID_M" in dset.variables:
+        height = "ZGRID_M"
+    mid_layer_height = np.array(dset[height])  # height in the layer upper interface of each layer
```

Comment:
This line would error with `height` unset, but you could instead add an `else:` and raise your own exception, to make it more clear.

Reply:
Just added it